blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala

```diff
@@ -23,6 +24,8 @@ private[http4s] class Http4sWSStage[F[_]](
 )(implicit F: ConcurrentEffect[F], val ec: ExecutionContext)
     extends TailStage[WebSocketFrame] {
 
+  private[this] val writeSemaphore = F.toIO(Semaphore[F](1L)).unsafeRunSync()
```

Comment:
I'm guessing this is being created and run here to make sure you only have 1 permit?

Reply:
Yes. I looked at passing it in, but we'd lose the guarantee that it's a single permit, and it would have required a MiMa exception that I didn't want to add.